pvlib/temperature.py

```diff
@@ -457,6 +466,119 @@ def faiman(poa_global, temp_air, wind_speed=1.0, u0=25.0, u1=6.84):
+        determined by Faiman for 7 silicon modules
+        in the Negev desert on an open rack at 30.9° tilt.
+        :math:`\left[\frac{\text{W}/{\text{m}^2}}{\text{C}}\right]`
+
+    u1 : numeric, default 6.84
+        Combined heat loss factor influenced by wind. The default value is one
+        determined by Faiman for 7 silicon modules
+        in the Negev desert on an open rack at 30.9° tilt.
+        :math:`\left[ \frac{\text{W}/\text{m}^2}{\text{C}\ \left( \text{m/s} \right)} \right]`
+
+    sky_view : numeric, default 1.0
+        Effective view factor limiting the radiative exchange between the
+        module and the sky. For a tilted array the expressions
+        (1 + 3*cos(tilt)) / 4 can be used as a first estimate for sky_view
+        as discussed in [4]_. The default value is for a horizontal module.
+        [unitless]
+
+    emissivity : numeric, default 0.88
+        Infrared emissivity of the module surface facing the sky. The default
+        value represents the middle of a range of values found in the
+        literature. [unitless]
+
+    Returns
+    -------
+    numeric, values in degrees Celsius
+
+    Notes
+    -----
+    All arguments may be scalars or vectors. If multiple arguments
+    are vectors they must be the same length.
+
+    When only irradiance, air temperature and wind speed inputs are provided
+    (`ir_down` is `None`) this function calculates the same device temperature
+    as the original faiman model. When down-welling long-wave radiation data
+    are provided as well (`ir_down` is not None) the default u0 and u1 values
+    from the original model should not be used because a portion of the
+    radiative losses would be double-counted.
+
+    References
+    ----------
+    .. [1] Faiman, D. (2008). "Assessing the outdoor operating temperature of
+       photovoltaic modules." Progress in Photovoltaics 16(4): 307-315.
+       :doi:`10.1002/pip.813`
+
+    .. [2] "IEC 61853-2 Photovoltaic (PV) module performance testing and energy
+       rating - Part 2: Spectral responsivity, incidence angle and module
+       operating temperature measurements". IEC, Geneva, 2018.
+
+    .. [3] "IEC 61853-3 Photovoltaic (PV) module performance testing and energy
+       rating - Part 3: Energy rating of PV modules". IEC, Geneva, 2018.
+
+    .. [4] Driesse, A. et al (2022) "Improving Common PV Module Temperature
+       Models by Incorporating Radiative Losses to the Sky". SAND2022-11604.
+       :doi:`10.2172/1884890`
+
+    See also
+    --------
+    pvlib.temperature.faiman
+
+    '''  # noQA: E501
+
+    # Contributed by Anton Driesse (@adriesse), PV Performance Labs. Nov., 2022
+
+    u0 = np.asanyarray(u0)
+    u1 = np.asanyarray(u1)
+    emissivity = np.asanyarray(emissivity)
+
+    abs_zero = np.array(-273.15)
+    sigma = np.array(scipy.constants.Stefan_Boltzmann)
+
+    if ir_down is None:
+        qrad_sky = np.array(0.0)
```

Comment:
```suggestion
    poa_global = np.asanyarray(poa_global)
    temp_air = np.asanyarray(temp_air)
    wind_speed = np.asanyarray(wind_speed)
    ir_down = np.asanyarray(ir_down)
    u0 = np.asanyarray(u0)
    u1 = np.asanyarray(u1)
    sky_view = np.asanyarray(sky_view)
    emissivity = np.asanyarray(emissivity)

    abs_zero = -273.15
    sigma = scipy.constants.Stefan_Boltzmann

    if ir_down is None:
        qrad_sky = 0.0
```

I see that the usage of np.asanyarray is very carefully chosen such that all inputs can be passed as lists and directly or indirectly be converted to arrays. The indirect conversion to arrays (by multiplying a list something array-like) seems undesirable and unintuitive to the user. Why not instead just convert all the inputs to array-like in the beginning and be straightforward about it?

Reply:
On the whole I believe this function behaves like numpy would.  The reason for the indirect and partial approach is to allow pandas and xarray data structures to be propagated from the first four arguments.  Try it, you might like it!

Reply:
What about this function requires any of the `np.asanyarray` calls?

Reply:
Ah, now I see the hidden conversation about allowing `list` input. That is not needed and comes at a cost of disallowing arbitrary array-like input that doesn't fully support the asanyarray machinery (e.g. dask, or whatever the pydata library of the month is).

Reply:
Could you elaborate on what doesn't work here?

Reply:
Overall I'm getting more push-back than expected on something which I though was a very simple technique to make the function very flexible.  If there is a concrete down-side I'd like to learn about it, especially the risks associated with ```asanyarray``` (which could be replaced with ```asarray``` in this particular function).  I've use this technique before and am likely to use it again unless there really is a problem with it.

Reply:
The function is already very flexible because you did a good job writing it :). It doesn't need `asanyarray`! Here's a concrete downside for `asanyarray`:

```python
In [1]: import numpy as np

In [2]: import dask.array as da

In [3]: da.from_array([0, 1])
Out[3]: dask.array<array, shape=(2,), dtype=int32, chunksize=(2,), chunktype=numpy.ndarray>

In [4]: np.asanyarray(da.from_array([0, 1]))
Out[4]: array([0, 1])

In [5]: np.asarray(da.from_array([0, 1]))
Out[5]: array([0, 1])
```

Reply:
> `np.asanyarray` is not robust.

I am curious to learn more about this...